app/controllers/users_controller.rb

```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  before_action :authorize_request, only: [:show, :destroy, :update]
+
+  def show
+    render json: @current_user.condensed, status: 200 
+  end
+
+  def create
+    user = User.new(user_params)
+    begin  
+      user.save!
+      @current_user = user 
+      token = JsonWebToken.encode( {user_id: @current_user.id} )
+    rescue ActiveRecord::RecordInvalid 
```

Comment:
You can do this but its a bit against the "rails way". Typically you should just call `.save` instead of `.save!` since it doesn't raise an error. .save will return false so instead of rescuing an error, you can just do `if @user.save`. However, this should work so I wont force you to change it.